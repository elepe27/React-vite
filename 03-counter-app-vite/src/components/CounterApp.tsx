import { props } from "../Interfaces/propsCounterApp";
import PropTypes from 'prop-types'
import { useState } from 'react';




export const CounterApp = ({value} :props) => {
    const [counter, setcounter] = useState(value);

    const handleAdd = ()=> setcounter(counter+1);
    const hanldeSubstract = ()=> setcounter(counter -1);
    const handleReset = ()=> setcounter(value);

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{ counter }</h2>
            <hr/>
            <button 
                onClick={ handleAdd } > +1</button>

            <button 
                aria-label="btn-reset"
                onClick={ handleReset } >Reset</button>
            <button 
                onClick={ hanldeSubstract } > -1</button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}



