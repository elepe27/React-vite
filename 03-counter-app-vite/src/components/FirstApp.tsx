import { props } from "../Interfaces/propsSaludoApp";
import PropTypes from 'prop-types';

/**Componente principal
 * @param arg:object
 */
export const FirstApp = ( {title, subtitle, edad , name} : props ) => {

    return (
        <>
            <h1 data-testid='test-title'>{ title }</h1>
            {/* <code> { JSON.stringify( nombre, 0, null, 4 ) } </code> */}
            <h2>{ subtitle }</h2>
            <h2>{ edad }</h2>
            <h2>{ name }</h2>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
    edad: 28,
    name: 'Esteban Lepe',
    title: 'No hay Titulo',
}