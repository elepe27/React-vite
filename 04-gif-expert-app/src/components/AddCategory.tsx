import { useState } from 'react';
import { IMyProps } from '../Interface/AddCategory/IAddCategoryProps';

export const AddCategory:React.FC<IMyProps> = ( { onNewCategory }:IMyProps ) => {

    /**Estado inicial de nuestro componente */
    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target: { value } }: any) => setInputValue(value);

    const onSubmit = ( e :any )=>{
        e.preventDefault();

        if (inputValue.trim().length <= 1) return;
            
        onNewCategory(inputValue);
        setInputValue('');
       
    }

    return (
        <form onSubmit={ onSubmit }>
            <input
                type='text'
                placeholder='Buscar gifs'
                value={ inputValue }
                onChange={ onInputChange }
                
            />
        </form>
    )
}