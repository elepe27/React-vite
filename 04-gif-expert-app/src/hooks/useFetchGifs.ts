import { useState, useEffect } from 'react';
import { getGifs } from '../Helpers/getGifs';


export const useFetchGifs = ( category:any ) => {


    const [images, setImages] = useState([]);
    const [isLoading, setisLoading] = useState( true )


    const getImages = async () => {
        const newImages: any = await getGifs(category);
        setImages(newImages);
        setisLoading( false );
        
    }

    useEffect(() => {
        getImages();

    }, [])



    return {
        images,
        isLoading
    }


}
