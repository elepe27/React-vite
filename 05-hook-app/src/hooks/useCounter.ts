import { useState } from 'react';


export const useCounter = ( initialValue = 10) => {

    const [counter, setCounter] = useState( initialValue );



    const onAddCounter = ( value = 1)=> setCounter( counter + value);
    const onDecrement = ( value = 1) => {
        if(counter === 0 ) return;
        setCounter ( counter - value)
    };
    const onReset = ( ) => setCounter (initialValue);

    

    
    
    return {
        counter,
        onAddCounter,
        onDecrement,
        onReset

    }
}
