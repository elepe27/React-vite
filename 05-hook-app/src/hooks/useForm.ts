import { useState } from 'react';


export const useForm = ( initialForm: object) => {


  const [ formState, setFormState] = useState( initialForm );



  const onInputChange = ({ target }: any) => {
    
    setFormState({
      ...formState,
      [target.name]: target.value
    });
  }

  const onReset = ()=> setFormState( initialForm );


  return {
    ...formState,
    formState ,
    onInputChange,
    onReset
  }

}
