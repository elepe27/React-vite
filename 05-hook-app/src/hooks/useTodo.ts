import { useReducer, useEffect } from 'react';
import { todoReducer } from '../08-useReducer/todoReducer';


const init = ()=>JSON.parse(localStorage.getItem('todo')|| '[]') || [];

export const useTodo = () => {

    const [todos, dispatchTodo]:any = useReducer(todoReducer, [], init);

    useEffect(()=> localStorage.setItem('todo',JSON.stringify(todos)),[ todos ])


    const onNewTodo = ( todo :any)=> {
        
        const action = {
            type:'[TODO] Add Todo',
            payload: todo
        }

        dispatchTodo( action );
    }

    const onDelete = ( id:any )=> {
        const action ={
            type: '[TODO] Remove Todo',
            payload: id
        }

        dispatchTodo( action );
    }

    const onToggleTodo = ( id:any )=> {
        const action ={
            type: '[TODO] Toggle',
            payload: id
        }

        dispatchTodo( action );
    }

    return {
        todos,
        onNewTodo,
        onDelete,
        onToggleTodo
        
    }


}
