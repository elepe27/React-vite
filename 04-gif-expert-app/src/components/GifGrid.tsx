import PropTypes from 'prop-types'

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }: any) => {


    const { images, isLoading } = useFetchGifs(category);

    return (

        <>

            {
                isLoading
                    ?
                    <p>Loading </p>

                    :
                    <>

                        <h3> {category} </h3>
                        <div className='card-grid'>
                            {images.map((img: any) => <GifGridItem {...img} key={img.id} />)}
                        </div>

                    </>
            }

        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.any.isRequired
}