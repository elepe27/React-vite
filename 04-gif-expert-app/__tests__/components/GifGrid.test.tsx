import { render, screen } from '@testing-library/react';
import { GifGrid } from '../../src/components/GifGrid';
import { useFetchGifs } from '../../src/hooks/useFetchGifs';

jest.mock('../../src/hooks/useFetchGifs');


describe('Pruebas en nuestro componente <GifGrid/>', () => {

    const category: string = 'Goku';

    test('el componente se debe renderizar', () => {
        useFetchGifs.mockReturnValue({
            images: [],
            isLoading:true

        });

        const { container } = render(<GifGrid category={category} />);

        expect(container).toMatchSnapshot();



    });

    test('debe de mostrar el loading', () => {

        


        render(<GifGrid category={category} />);
        expect(screen.getByText('Loading'));


    });

    // test('debe de mostrar items, cuando se cargan las imagenes', () => {

    //     const gifs = [{
    //         id:'ABC',
    //         title:'Goku',
    //         url:'https://asdf.jpg'
    //     },
    //     {
    //         id:'123',
    //         title:'Luffy',
    //         url:'https://gomugomugo.jpg'
    //     }];

    //     useFetchGifs.mockReturnValue({
    //         images: gifs,
    //         isLoading:false

    //     });

    //     screen.debug();
    //     expect( screen.getAllByRole('img').length).toBe(2);


    // });


});