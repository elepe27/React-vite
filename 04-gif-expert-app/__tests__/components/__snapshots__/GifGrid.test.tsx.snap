// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en nuestro componente <GifGrid/> el componente se debe renderizar 1`] = `
<div>
  <p>
    Loading 
  </p>
</div>
`;
